service/history/api/describeworkflow/api.go

```diff
@@ -192,25 +192,16 @@ func Invoke(
 		}
 	}
 
-	if executionInfo.CloseVisibilityTaskCompleted {
-		// If close visibility task has completed, then search attributes and memo
-		// were removed from mutable state, and we need to fetch from visibility.
-		visResponse, err := persistenceVisibilityMgr.GetWorkflowExecution(
-			ctx,
-			&manager.GetWorkflowExecutionRequest{
-				NamespaceID: namespaceID,
-				Namespace:   namespace.Name(req.Request.GetNamespace()),
-				RunID:       executionState.RunId,
-				WorkflowID:  executionInfo.WorkflowId,
-				CloseTime:   executionInfo.CloseTime,
-			},
-		)
-		if err != nil {
-			return nil, err
-		}
-		result.WorkflowExecutionInfo.SearchAttributes = visResponse.Execution.SearchAttributes
-		result.WorkflowExecutionInfo.Memo = visResponse.Execution.Memo
+	memo, err := mutableState.GetMemo(ctx, persistenceVisibilityMgr)
+	if err != nil {
+		return nil, err
+	}
+	searchAttributes, err := mutableState.GetSearchAttributes(ctx, persistenceVisibilityMgr)
+	if err != nil {
+		return nil, err
```

Comment:
Wondering if we should create a function to return both memo and search attributes to avoid making two calls to visibility store.
cc: @yiminc @alexshtin

Reply:
That's what I had originally but @yux0 suggested we add individual methods for each field. I think it should be one call though, like GetDynamicFields

Reply:
But still a method on the mutable state like YX suggested

Reply:
I don't mind to be one method like GetMemoAndSearchAttributes in mutable state.

Reply:
Ok consolidated in one method.